table/block_based/filter_policy.cc

```diff
@@ -756,4 +803,50 @@ FilterBuildingContext::FilterBuildingContext(
 
 FilterPolicy::~FilterPolicy() { }
 
+static bool LoadFilterPolicy(const std::string& id,
+                             std::shared_ptr<FilterPolicy>* policy) {
+  bool success = true;
+  if (id == "rocksdb.BuiltinBloomFilter") {
+    policy->reset(
+        new BloomFilterPolicy(0, BloomFilterOptions::kDeprecatedBlock));
+  } else {
+    success = false;
+  }
+  return success;
+}
+
+Status FilterPolicy::CreateFromString(
+    const std::string& value, const ConfigOptions& opts,
+    std::shared_ptr<const FilterPolicy>* policy) {
+  const std::string kBloomName = "bloomfilter:";
+  Status s;
+  if (value == kNullptrString) {
+    policy->reset();
+#ifndef ROCKSDB_LITE
+  } else if (value.compare(0, kBloomName.size(), kBloomName) == 0) {
+    size_t pos = value.find(':', kBloomName.size());
```

Comment:
Here the string representation is the same as the already public string representation from ParseBlockBasedTableOption, but can you explain the mapping of "rocksdb.BuiltinBloomFilter" -> new BloomFilterPolicy(0, BloomFilterOptions::kDeprecatedBlock) in LoadFilterPolicy?

Reply:
- The "value" can look like the old string (bloomfilter:*), in which case it is handled like before;
- If the "value" is "rocksdb.BuiltinBloomFilter" or contains "id=rocksdb.BuiltinBloomFilter", we care a BloomFilterPolicy.
- If the value contains other properties (bits_per_key, mode), those are set in the FilterPolicy (via ConfigureFromMap/String).
- After all of the properties are configured, the PrepareOptions method is invoked to finish the setup of the FilterPolicy. 

A fully qualified BloomFilterPolicy would now look something like "{id=rocksdb.BuiltinBloomFilter; bits_per_key=10; mode=auto;}".  This representation matches how all of the other properties are represented in the options file.